cpt/builds_generator.py

```diff
@@ -219,39 +219,32 @@ def get_visual_builds_for_version(visual_runtimes, visual_version, arch, shared_
                 "arch": arch}
     sets = []
 
-    if shared_option_name:
-        if "MT" in visual_runtimes and "Release" in build_types:
-            sets.append(({"build_type": "Release", "compiler.runtime": "MT"},
-                         {shared_option_name: False}, {}, {}))
-            if dll_with_static_runtime:
-                sets.append(({"build_type": "Release", "compiler.runtime": "MT"},
-                             {shared_option_name: True}, {}, {}))
-        if "MTd" in visual_runtimes and "Debug" in build_types:
-            sets.append(({"build_type": "Debug", "compiler.runtime": "MTd"},
-                                  {shared_option_name: False}, {}, {}))
-            if dll_with_static_runtime:
-                sets.append(({"build_type": "Debug", "compiler.runtime": "MTd"},
-                                      {shared_option_name: True}, {}, {}))
-        if "MD" in visual_runtimes and "Release" in build_types:
-            sets.append(({"build_type": "Release", "compiler.runtime": "MD"},
-                                  {shared_option_name: False}, {}, {}))
-            sets.append(({"build_type": "Release", "compiler.runtime": "MD"},
-                                  {shared_option_name: True}, {}, {}))
-        if "MDd" in visual_runtimes and "Debug" in build_types:
-            sets.append(({"build_type": "Debug", "compiler.runtime": "MDd"},
-                                  {shared_option_name: False}, {}, {}))
-            sets.append(({"build_type": "Debug", "compiler.runtime": "MDd"},
-                                  {shared_option_name: True}, {}, {}))
-
-    else:
-        if "MT" in visual_runtimes and "Release" in build_types:
-            sets.append(({"build_type": "Release", "compiler.runtime": "MT"}, {}, {}, {}))
-        if "MTd" in visual_runtimes and "Debug" in build_types:
-            sets.append(({"build_type": "Debug", "compiler.runtime": "MTd"}, {}, {}, {}))
-        if "MDd" in visual_runtimes and "Debug" in build_types:
-            sets.append(({"build_type": "Debug", "compiler.runtime": "MDd"}, {}, {}, {}))
-        if "MD" in visual_runtimes and "Release" in build_types:
-            sets.append(({"build_type": "Release", "compiler.runtime": "MD"}, {}, {}, {}))
+    def append_to_sets(runtimes, builds, check_dll_with_static_runtime):
+        for rt in runtimes:
```

Comment:
Why not iterate directly `visual_runtimes` am I missing something? if it is an input check we could do it before in a separated function? Same for `build_types`.

Reply:
Yes, it is possible to iterate over visual_runtimes. 

I came up with a simpler loop. But the new loop changes the order of the output. I need to change the order in the `expected` array in the unit tests.